runtime-components/non-repudiation-client/src/test/scala/com/daml/nonrepudiation/client/SigningInterceptorSpec.scala

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation.client
+
+import java.time.{Clock, Instant, ZoneId}
+import java.util.Collections
+
+import com.daml.ledger.api.v1.CommandServiceOuterClass.SubmitAndWaitRequest
+import com.daml.ledger.javaapi.data.Command
+import com.daml.ledger.api.v1.CommandsOuterClass.{Command => ProtoCommand}
+import com.daml.ledger.api.v1.command_service.CommandServiceGrpc.CommandService
+import com.daml.ledger.api.v1.command_submission_service.CommandSubmissionServiceGrpc.CommandSubmissionService
+import com.daml.ledger.rxjava.DamlLedgerClient
+import com.daml.nonrepudiation.testing._
+import com.daml.nonrepudiation.{
+  AlgorithmString,
+  CommandIdString,
+  NonRepudiationProxy,
+  SignatureBytes,
+  SignedPayload,
+}
+import com.daml.resources.grpc.{GrpcResourceOwnerFactories => Resources}
+import io.grpc.netty.NettyChannelBuilder
+import org.scalatest.Inside
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.DurationInt
+
+final class SigningInterceptorSpec extends AsyncFlatSpec with Matchers with Inside {
+
+  behavior of "SigningInterceptor"
+
+  it should "be possible to use it with the Java bindings" in {
+    val (key, certificate) = generateKeyAndCertificate()
+    val builders =
+      DummyTestSetup.Builders(
+        useNetworkStack = true,
+        serviceExecutionContext = ExecutionContext.global,
+      )
+
+    val certificates = new Certificates
+    val signedPayloads = new SignedPayloads[CommandIdString]
+
+    val expectedAlgorithm = AlgorithmString.SHA256withRSA
+    val expectedFingerprint = certificates.put(certificate)
+    val expectedTimestamp = Instant.now()
+
+    val proxy =
+      for {
+        _ <- Resources.forServer(builders.participantServer, 5.seconds)
+        participant <- Resources.forChannel(builders.participantChannel, 5.seconds)
+        proxy <- NonRepudiationProxy.owner(
+          participant,
+          builders.proxyServer,
+          certificates,
+          signedPayloads,
+          Clock.fixed(expectedTimestamp, ZoneId.systemDefault()),
+          CommandService.scalaDescriptor.fullName,
+          CommandSubmissionService.scalaDescriptor.fullName,
+        )
+      } yield proxy
+
+    proxy.use { _ =>
+      val clientChannelBuilder =
+        builders.proxyChannel
+          .asInstanceOf[NettyChannelBuilder]
+          .intercept(new SigningInterceptor(key, certificate))
```

Comment:
What happens if I use something other than the command service? From a quick look, I couldn’t find anything that prevents the interceptor from kicking in. iirc we only insert it for command services on the non-repudation server side so it probably doesn’t break anything but unnecessarily signing things also doesn’t seem that great of an idea.

At the very least, I think we should have a test for accessing other services (e.g., one of acs, transaction service, …). Totally fine to do this in another PR.

Reply:
Very good point, thanks. Probably we actually need something specific.

Reply:
I've addressed this with https://github.com/digital-asset/daml/pull/9050. After that is merged, I can rebase this on `main`, limit signing to commands and add a test for this.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/9036/commits/dec170bad4fb2d3a790723ea5343d0f3acba08d3